agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -3577,7 +3577,7 @@ bool slave_thread::autoconfig_wsc_authenticate(std::shared_ptr<ieee1905_1::tlvWs
 bool slave_thread::autoconfig_wsc_parse_m2_encrypted_settings(
     std::shared_ptr<ieee1905_1::tlvWscM2> m2, uint8_t authkey[32], uint8_t keywrapkey[16],
     std::string &ssid, sMacAddr &bssid, WSC::eWscAuth &auth_type, WSC::eWscEncr &encr_type,
-    bool &backhaul, bool &fronthaul, bool &teardown)
+    bool &backhaul, bool &fronthaul, bool &teardown, std::shared_ptr<WSC::cConfigData> &credentials)
```

Comment:
> Don't we have the backhaul, fronthaul and teardown bits as part of `cConfigData`?

In fact, *all* the credentials are in cConfigData; we don't need ssid, bssid, auth_type, encr_type either. However, the backhaul, fronthaul and teardown bits are a bit difficult to retrieve since they're part of the vendor extension attribute.

I originally proposed to use cConfigData, but I'm thinking that maybe we should define a new, cleaner structure instead...

Anyway, I'm OK with using cConfigData for now.

Reply:
Just to clarify: I agree with @tomereli that it doesn't make sense to return anything other than `credentials`.

Another thing, however: instead of a shared_ptr, it should be a reference. A reference makes it clear that the function will not keep the object alive after it returns. Cfr. [CppCoreGuildeline R.30](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rr-smartptrparam).

Reply:
Removed ssid, bssid, auth_type, encr_type. Backhaul, fronthaul and teardown left as is because they should be parsed as bit map.
It should be shared_ptr because inside tlvf it is shared ptr. Could change yaml to have it as reference.